### PR TITLE
test: add tests to PauseUnpauseTokenAccountPrecompileSuite

### LINKED ISSUE #12490: Implement negative test cases for pause/unpause

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

add tests to PauseUnpauseTokenAccountPrecompileSuite

**Related issue(s)**:

Fixes #12490 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PauseUnpauseTokenAccountPrecompileSuite.java [hunk 1]
```diff
@@ -88,7 +113,263 @@ protected Logger getResultsLogger() {
...
     @Override
     public List<HapiSpec> getSpecsInSuite() {
-        return List.of(noTokenIdReverts(), noAccountKeyReverts());
+        return List.of(
+                noTokenIdReverts(),
+                noAccountKeyReverts(),
+                pauseFungibleTokenHappyPath(),
+                unpauseFungibleTokenHappyPath(),
+                pauseNonFungibleTokenHappyPath(),
+                unpauseNonFungibleTokenHappyPath());
```
Comment (MrValioBg) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PauseUnpauseTokenAccountPrecompileSuite.java:122]: The test names ending in "HappyPath" may suggest that they cover only positive scenarios.
For clarity and better understanding, maybe we could consider more descriptive names that highlight the scenarios being tested.

**Review by MrValioBg — APPROVED** [review]
LGTM!

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenPauseHandler.java [hunk 2]
```diff
@@ -79,6 +81,7 @@ public void handle(@NonNull final HandleContext context) {
         var token = tokenStore.get(op.tokenOrElse(TokenID.DEFAULT));
         validateTrue(token != null, INVALID_TOKEN_ID);
         validateTrue(token.hasPauseKey(), TOKEN_HAS_NO_PAUSE_KEY);
+        validateFalse(token.deleted(), TOKEN_WAS_DELETED);
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenPauseHandler.java:84]: Nice catch! 🙌

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @mustafauzunn !

**Pull request merged**